### PR TITLE
Add support for post requests and refactor service store mapping models

### PR DESCRIPTION
**Pull request by hardikmaheshwari**
## Summary
These changes will let users fetch data from post api's by generating request bodies at runtime.
These changes also refactor the current ServiceStore mapping models to better align with the concepts.
Depends on - https://github.com/finos/legend-engine/pull/580

## How did you test this change?

- [X] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

**Comment by hardikmaheshwari** [comment]
@MauricioUyaguari @akphi Can you please review these changes?
Roundtrip tests will be fixed once attached engine PR is merged

File: packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts [hunk 1]
```diff
@@ -16,30 +16,24 @@
...
 import type { ServiceStoreService } from '../model/ServiceStoreService';
 import type { RootServiceInstanceSetImplementation } from './RootServiceInstanceSetImplementation';
+import type { ServiceRequestBuildInfo } from './ServiceRequestBuildInfo';
 
 export class ServiceMapping implements Hashable {
   owner!: RootServiceInstanceSetImplementation;
   service!: ServiceStoreService;
-  parameterMappings: ServiceParameterMapping[] = [];
-  path: object | undefined; // @MARKER GENERATED MODEL DISCREPANCY --- Studio does not process lambda
+  pathOffset: object | undefined; // @MARKER GENERATED MODEL DISCREPANCY --- Studio does not process lambda
```
Comment (akphi) [inline packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts:28]: -> `?:`

File: packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts [hunk 2]
```diff
@@ -16,30 +16,24 @@
...
+  requestBuildInfo: ServiceRequestBuildInfo | undefined;
```
Comment (akphi) [inline packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts:29]: -> `?:`

File: packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts (hunk 1 above)
Comment (akphi) [inline packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts:28]: -> `?:`

File: packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts (hunk 2 above)
Comment (akphi) [inline packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceMapping.ts:29]: -> `?:`

File: packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceRequestBuildInfo.ts [hunk 3]
```diff
@@ -15,35 +15,19 @@
...
-import { ServiceParameterMapping } from './ServiceParameterMapping';
-import { action, computed, makeObservable, observable } from 'mobx';
 
-export class PropertyIndexedParameterMapping
-  extends ServiceParameterMapping
-  implements Hashable
-{
-  property!: string;
+export class ServiceRequestBuildInfo implements Hashable {
+  requestBodyBuildInfo: ServiceRequestBodyBuildInfo | undefined;
```
Comment (akphi) [inline packages/legend-extension-external-store-service/src/models/metamodels/pure/model/packageableElements/store/serviceStore/mapping/ServiceRequestBuildInfo.ts:23]: Let's use `?:` as its more correct, the value sent by engine is optional or `undefined`

File: packages/legend-extension-external-store-service/src/models/protocols/pure/v1/model/packageableElements/store/serviceStore/mapping/V1_ServiceParameterMapping.ts [hunk 4]
```diff
@@ -16,6 +16,10 @@
 
 import type { Hashable } from '@finos/legend-shared';
 
+/**
+ * TODO: this will be removed in upcoming releases
+ * @deprecated
+ */
```
Comment (akphi) [inline packages/legend-extension-external-store-service/src/models/protocols/pure/v1/model/packageableElements/store/serviceStore/mapping/V1_ServiceParameterMapping.ts:22]: We probably should mention in the changeset that we `deprecate` these.

**Review by akphi — CHANGES_REQUESTED** [review]

**Comment by hardikmaheshwari** [comment]
> @hardikmaheshwari yup, let's switch it to `non draft` next time 😛

@akphi I marked it as draft as I don't want this to be merged before engine PR's 
Thanks for reviewing

**Review by akphi — APPROVED** [review]
I'll remove the deprecated models actually. `studio` will quietly ignore ignored properties while parsing anyway.

**Pull request merged**